og_access/og_access.module

```diff
@@ -0,0 +1,208 @@
+<?php
+
+/**
+ * @file
+ * Enable access control for private and public groups and group content.
+ *
+ * @TODO handle visibility change
+ * @TODO set group content visibility default to that of the group
+ * @TODO move grants/access to service
+ */
+
+use Drupal\Core\Config\Entity\ConfigEntityBundleBase;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\field\Entity\FieldConfig;
+use Drupal\node\NodeInterface;
+use Drupal\og\Og;
+
+/**
+ * The access realm of group member.
+ */
+define('OG_ACCESS_REALM', 'og_access');
+
+/**
+ * Group public access field.
+ */
+define('OG_ACCESS_FIELD', 'group_access');
+
+/**
+ * Group public access field.
+ */
+define('OG_CONTENT_ACCESS_FIELD', 'group_content_access');
+
+/**
+ * Public group/group content access.
+ */
+define('OG_ACCESS_PUBLIC', 0);
+
+/**
+ * Private group/group content access.
+ */
+define('OG_ACCESS_PRIVATE', 1);
+
+/**
+ * Implements hook_node_grants().
+ */
+function og_access_node_grants(AccountInterface $account, $op) {
+  if ($op != 'view') {
+    return [];
+  }
+
+  /** @var \Drupal\og\MembershipManager $membership_manager */
+  $membership_manager = \Drupal::service('og.membership_manager');
+  if ($groups = $membership_manager->getUserGroups($account)) {
+    foreach ($groups as $group_type => $entity_groups) {
+      /** @var \Drupal\core\Entity\EntityInterface $group */
+      foreach ($entity_groups as $group) {
+        $realm = OG_ACCESS_REALM . ':' . $group_type;
+        $grants[$realm][] = $group->id();
+      }
+    }
+  }
+
+  return !empty($grants) ? $grants : [];
+}
+
+/**
+ * Implements hook_node_access_records().
+ */
+function og_access_node_access_records(NodeInterface $node) {
+  if (empty($node->status)) {
```

Comment:
`!$node->isPublished()`

Reply:
It's already fixed in the fork. I've only fixed a small language issue in the comment below this line.